### PR TITLE
Added deriv_root argument to CLI

### LINKED ISSUE #772: Add command line option for deriv_root
As I develop a Docker App using mne-bids-pipeline, I want a straightforward way to overwrite the bids_root and deriv_root values defined in the configuration file from the command line. This enhancement aims to simplify Docker volume mapping.

Currently, the option to override `bids_root `is already available ([source](https://github.com/mne-tools/mne-bids-pipeline/blob/ea95979fdb0ef807b3cd262da409cf010ada3da3/mne_bids_pipeline/_main.py#L56)), but there is no similar option for `deriv_root`.

I suggest adding a command-line option to the mne-bids-pipeline tool to allow users to specify the deriv_root directory. This behavior is similar to other BidsApps such as `FMRIprep `and `MRIQC`, where command line arguments overwrite configuration files.

If you are interested, I would be happy to contribute by taking care of the pull request.

### PR DESCRIPTION
**Pull request by vferat**
Closes #772.

I do not see unit test and/or test for other cli arguments, so I didn't implemented any for the current changes. Let me know if you think it is necessary.

### Before merging …
- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: mne_bids_pipeline/_config.py [hunk 1]
```diff
@@ -217,7 +217,7 @@
     ```
 """
 
-data_type: Optional[Literal["meg", "eeg"]] = None
+data_type: Optional[Literal["meg", "eeg"]] = "eeg"
```
Comment (larsoner) [inline mne_bids_pipeline/_config.py:220]: Same

**Comment by larsoner** [comment]
Argh I enabled pre-commit.ci and it pushed something it shouldn't have... Feel free to keep working on this and don't worry about `freesurfer` code changes. I'll fix those in a separate PR then push a revert commit here. We squash history when merging later anyway so it should be okay :)

File: mne_bids_pipeline/_main.py [hunk 2]
```diff
@@ -148,6 +161,8 @@ def main():
     overrides = SimpleNamespace()
     if root_dir:
         overrides.bids_root = pathlib.Path(root_dir).expanduser().resolve(strict=True)
+    if deriv_root:
+        overrides.deriv_root = pathlib.Path(deriv_root).expanduser().resolve(strict=False)
```
Comment (larsoner) [inline mne_bids_pipeline/_main.py:165]:
```
mne_bids_pipeline/_main.py:165:89: E501 Line too long (90 > 88 characters)
```

File: mne_bids_pipeline/_main.py [hunk 3]
```diff
@@ -59,6 +59,18 @@ def main():
...
     )
+    parser.add_argument(
+        "--deriv_root",
+        dest="deriv_root",
+        default=None,
+        help=dedent(
+            """\
+        The root of the derivatives directory
+        in which the pipeline will store the processing results.
+        If None, this will be derivatives/mne-bids-pipeline
```
Comment (hoechenberger) [inline mne_bids_pipeline/_main.py:70]:
```suggestion
        If unspecified, this will be derivatives/mne-bids-pipeline
```

**Pull request merged**